spectrum-offchain/src/executor.rs

```diff
@@ -0,0 +1,111 @@
+use std::fmt::Display;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use ergo_lib::chain::transaction::Transaction;
+use futures::{stream, Stream, StreamExt};
+use log::warn;
+use parking_lot::Mutex;
+use type_equalities::{trivial_eq, IsEqual};
+
+use crate::backlog::Backlog;
+use crate::box_resolver::BoxResolver;
+use crate::data::unique_entity::{Predicted, Traced};
+use crate::data::{OnChainEntity, OnChainOrder};
+use crate::network::ErgoNetwork;
+
+/// Indicated the kind of failure on at attempt to execute an order offline.
+pub enum RunOrderFailure<TOrd: OnChainOrder> {
+    /// Discard order in the case of fatal failure.
+    Fatal(String, TOrd::TOrderId),
+    /// Return order in the case of non-fatal failure.
+    NonFatal(String, TOrd),
+}
+
+pub trait RunOrder<TEntity, TCtx>: OnChainOrder + Sized {
+    /// Try to run the given `TOrd` against the given `TEntity`.
+    /// Returns transaction and the next state of the persistent entity in the case of success.
+    /// Returns `RunOrderError<TOrd>` otherwise.
```

Comment:
Name mismatch with `RunOrderFailure`. Also `RunOrderError` is a more idiomatic Rust name, I think

Reply:
Fixed in the subsequent PR